common/src/main/java/org/opensearch/ml/common/PredictMode.java

```diff
@@ -0,0 +1,18 @@
+package org.opensearch.ml.common;
+
+import java.util.Locale;
+
+public enum PredictMode {
+    PREDICT,
+    BATCH,
+    ASYNC,
```

Comment:
Seems async should not be a mode. No matter predict or batch, they could possibly use async way

Reply:
Let's use a different name for asynchronously invoking a model for a single request. Async is only used by sageMaker for this type of request. May not need to support this type of invoke at all.